docker/compile/deprecated/Dockerfile.ubuntu16.04

```diff
@@ -49,6 +49,8 @@ RUN apt-get update && apt-get -y install \
 
 ENV JAVA_HOME /usr/lib/jvm/java-11-openjdk-amd64
 
+RUN update-alternatives --install /usr/bin/python python /usr/bin/python3 10
```

Comment:
this image also has python3.5, so may as well delete this too. I think if anyone needed an old base OS image, they'd probably be maintaining their own Dockerfile for that or similar purposes anyway

Reply:
I moved both 14.04 (Well past the End of Life) and 16.04 (about to reach End of Life) to the deprecated folder. I think they should be removed if they don't support Python 3.6. Both images won't build successfully.